src/openrct2-ui/WindowManager.cpp

```diff
@@ -646,6 +646,127 @@ class WindowManager final : public IWindowManager
         }
         return nullptr;
     }
+
+    /**
+     * Finds the first window with the specified window class.
+     *  rct2: 0x006EA8A0
+     * @param cls (cl) with bit 15 set
+     * @returns the window or NULL if no window was found.
+     */
+    WindowBase* FindByClass(WindowClass cls) override
+    {
+        for (auto& w : g_window_list)
+        {
+            if (w->flags & WF_DEAD)
+                continue;
+            if (w->classification == cls)
+            {
+                return w.get();
+            }
+        }
+        return nullptr;
+    }
+
+    /**
+     * Finds the first window with the specified window class and number.
+     *  rct2: 0x006EA8A0
+     * @param cls (cl) without bit 15 set
+     * @param number (dx)
+     * @returns the window or NULL if no window was found.
+     */
+    WindowBase* FindByNumber(WindowClass cls, rct_windownumber number) override
+    {
+        for (auto& w : g_window_list)
+        {
+            if (w->flags & WF_DEAD)
+                continue;
+            if (w->classification == cls && w->number == number)
+            {
+                return w.get();
+            }
+        }
+        return nullptr;
+    }
+
+    // TODO: Use variant for this once the window framework is done.
```

Comment:
We might want to open an issue for this.

Reply:
Heh, I didn't even introduce this todo. It must've been in there for years 😅

Reply:
Windows sometimes have data assigned such as the entity id or ride id or banner id something unique to the window. This is just the C way of storing such things.